### PR TITLE
[needs-docs] Rename "Create parallel lines" and "Array of offset feat…

### PR DESCRIPTION
**Pull request by DelazJ**
…ures" algorithms

See discussion at http://osgeo-org.1560.x6.nabble.com/QGIS-Developer-Interrogation-about-algorithms-to-offset-translate-features-td5374924.html

### CODE REVIEW

**Comment by alexbruy** [comment]
What about algorithms names? After this change they will be inconsistent and misleading

**Comment by DelazJ** [comment]
@alexbruy that would concern only the "Create parallel lines", right? Issue i mention line 29, suggesting to find a new name for the class (hence, the alg)... 
I have no inspiration for this so any proposal is welcome.

File: src/analysis/processing/qgsalgorithmarrayfeatures.cpp [hunk 1]
```diff
@@ -46,18 +46,18 @@ QString QgsArrayFeaturesAlgorithm::groupId() const
 
 QString QgsArrayFeaturesAlgorithm::outputName() const
 {
-  return QObject::tr( "Offsets" );
+  return QObject::tr( "Offsets" ); //looking for a label
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmarrayfeatures.cpp]: Can you remove the comments here?

File: src/analysis/processing/qgsalgorithmarrayfeatures.cpp [hunk 2]
```diff
@@ -26,12 +26,12 @@ QString QgsArrayFeaturesAlgorithm::name() const
...
 QString QgsArrayFeaturesAlgorithm::displayName() const
 {
-  return QObject::tr( "Array of offset features" );
+  return QObject::tr( "Array of translated features" );
 }
 
 QStringList QgsArrayFeaturesAlgorithm::tags() const
 {
-  return QObject::tr( "translate,duplicate,grid,spaced,moved,copy,features,objects,step,repeat" ).split( ',' );
+  return QObject::tr( "translate,parallel,duplicate,grid,spaced,moved,copy,features,objects,step,repeat" ).split( ',' );
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmarrayfeatures.cpp]: Also add "offset"

File: src/analysis/processing/qgsalgorithmparallellines.cpp [hunk 3]
```diff
@@ -26,7 +26,7 @@ QString QgsCreateParallelLinesAlgorithm::name() const
 
 QString QgsCreateParallelLinesAlgorithm::displayName() const
 {
-  return QObject::tr( "Create parallel lines" );
+  return QObject::tr( "Array of offset lines" ); // the class might need a new name. Ideas?
```
Comment (nyalldawson) [inline src/analysis/processing/qgsalgorithmparallellines.cpp]: Add "parallel" to tags, remove comment

**Comment by nirvn** [comment]
@DelazJ , I'm +1 on the array of offset features -> array of translated features renaming, but I'm not so sure about the create parallel lines -> array of offset lines change. I feel "parallel" might be a more meaningful name for a large amount of users.

Can you update the PR to rename only the first algorithm?

**Comment by DelazJ** [comment]
@nirvn Thanks for giving it a look
> I feel "parallel" might be a more meaningful name for a large amount of users.

Yes I agree with you but Offset is not that uncommon in QGIS (offset lines alg, offset in style properties, Offset curve digitizing tool or function...). There are many occurrences of this word.
Actually, the "create parallel lines" name annoys me more than the other. I don't feel it's the right name for what the alg does:
1. It exactly returns features I'd get if I iteratively run the "Offset lines" algorithm. So, shouldn't the names look alike?
1. the now named "array of translated features" imho really translates features and keeps the parallelism between them (you get the same geometry; only coordinates are offset by same value). I'd have named it "create parallel features" if I could consider two points as parallel. ;)
The "create parallel lines" otoh doesn't do that: it offsets the geometry, modifies it in most cases, leading to a new geometry that can be really different from the initial (you just need more than one segment in the feature, which is really common), with joint segments. Reason why I don't think that pushing "parallel" at the top of this alg is fair/right/what user would expect. Examples below
parallel_lines

**Comment by nyalldawson** [comment]
How about "array of offset (parallel) lines"?

**Comment by DelazJ** [comment]
Honestly i don't think that parallel has anything to do at this place (more than some other algs) and i even find it misleading. But... if i'm the only one with this idea, then maybe i'm missing some definition and I will follow what is decided.

**Comment by nyalldawson** [comment]
> Honestly i don't think that parallel has anything to do at this place (more than some other algs) and i even find it misleading. But... if i'm the only one with this idea, then maybe i'm missing some definition and I will follow what is decided.

I don't think we'll be able to resolve this question, it's too heavily language/culture dependent. I think just go with "array of offset (parallel) lines" and let's get it merged for 3.4.

**Pull request merged**